### ISSUE TITLE
Can I set different set offset based on the resource type?

### ISSUE THREAD

**Issue by daemyung** [KhronosGroup/glslang#2017]
My shader is below.

layout(binding = 0) uniform Colors {
	vec4 color0;
	vec4 color1;
	vec4 color2;
};

layout(binding = 1) uniform Geometry {
	vec4 translate;
	vec4 rotate;
	vec4 scale;
};

layout(binding = 0) uniform sampler2D base_texture0;
layout(binding = 1) uniform sampler2D base_texture1;
layout(binding = 2) uniform sampler2D base_texture2;

I don't use 'set=X' and I just want to give the different set based on the resource type? Is it possible now? If if's possible could you help me?

**Comment by johnkslang** [comment]
I don't believe we have an set shift functionality.  Are you sure you wouldn't want to shift the bindings, rather than the set?  I'm not an expert on set allocation, but understood it to be more carefully designed about what each level would mean, as in the numbers are less arbitrary.

**Comment by daemyung** [comment]
Yes I want a set shift functionality because I don't want to make any limitation of resources. What I mean that if I use a binding shift functionality the I need to make limitation how many binding will be shifted each resource. For example, my current engine just support to bind 8 uniform buffer objects so I shift 8 binding slot for textures. Later H/W is improved I can use 16 uniform buffer objects. Now I need to fix my code to shift 16 bindings for textures. If I can shift set index, I don't have to do it. Because the first set index is dedicated for uniform buffer object and the second set index is dedicated for textures.
